internal/util/tree-sitter.go

```diff
@@ -48,17 +48,26 @@ func GuessWithTreeSitter(ctx context.Context, dir string, lang *sitter.Language,
 	}
 
 	pathsToSearch := []string{}
-	for _, pattern := range searchGlobPatterns {
-		globSearchPaths, err := fs.Glob(dirFS, pattern)
+	err := fs.WalkDir(dirFS, ".", func(dir string, d fs.DirEntry, err error) error {
```

Comment:
smol caveat: this will be slow af in some situations, where there are massive directories (like `node_modules` and `.pythonlibs` and `venv`) with gajillion files that would potentially match. we may be okay if we `return fs.SkipDir` if we encounter those directories to keep things speedy.

Reply:
also can we have a smol test?

Reply:
So I was relying on the `ignored` infrastructure to resolve this, but you're right about if somebody does like a `mv node_modules node_modules.bak`. In doing some manual testing I discovered that the ignored stuff basically doesn't work anywhere near like what I expected, so I'm trying to figure out the best options we have. I'll add a test when I've got that sorted.

Reply:
OK! Ready for re-review